whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/MessageService.java

```diff
@@ -58,21 +59,24 @@ public EmailMessageInfo getAutomatedFailureMessageInfo(final MimeMessage message
         return autoSubmittedMessageParser.parse(message);
     }
 
-    public void verifyAndSetAsUndeliverable(final EmailMessageInfo message) {
-        final List<String> outgoingEmail = outgoingMessageDao.getEmails(message.messageId());
+    public void verifyAndSetAsUndeliverable(final EmailMessageInfo messageInfo) {
+        final List<String> outgoingEmail = outgoingMessageDao.getEmails(messageInfo.messageId());
 
-        if (!isValidMessage(message, outgoingEmail)) {
+        if (!isValidMessage(messageInfo, outgoingEmail)) {
             return;
         }
 
-        LOGGER.debug("Undeliverable message-id {} email {}", message.messageId(), StringUtils.join(message.emailAddresses(), ", "));
-        message.emailAddresses().forEach(email -> {
+        LOGGER.debug("Undeliverable message-id {} email {}", messageInfo.messageId(), StringUtils.join(messageInfo.emailAddresses(), ", "));
+        for (String email : messageInfo.emailAddresses()) {
             try {
-                emailStatusDao.createEmailStatus(email, EmailStatusType.UNDELIVERABLE);
+                emailStatusDao.createEmailStatus(email, EmailStatusType.UNDELIVERABLE, messageInfo.message());
             } catch (DuplicateKeyException ex) {
-                LOGGER.debug("Email already exist in EmailStatus table {}", StringUtils.join(message.emailAddresses(), ", "), ex);
+                LOGGER.debug("Email already exist in EmailStatus table {}", StringUtils.join(messageInfo.emailAddresses(), ", "), ex);
+            } catch (MessagingException | IOException e) {
+                LOGGER.error("Unable to transform the bounced message of {} into byte[]", email);
+                emailStatusDao.createEmailStatus(email, EmailStatusType.UNDELIVERABLE);
```

Comment:
Instead of null can you save the message that you are unable to parse?

Reply:
the data in database is a longblob type...
can I store a raw string message there? I thought that I need to get the byte[] to store it.